### PR TITLE
3/n pin Gemini dependency (google api)

### PR DESCRIPTION
**Pull request by Ankush-lastmile**
3/n pin Gemini dependency (google api)

## What

Pin google-generative ai, the library used for Gemini

## Why
Anticipiating potential updates to the library, its prudent to pin its current version.
Additionally, it ensures that all users operate on the same, specified version of the API.

## Testplan

I ran `pip list | grep google-generativeai` to find the version of the library that was installed locally.

```
ankush@ap-mbp python % pip list | grep google-generativeai
google-generativeai                           0.3.1
```

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/534).
* #558
* __->__ #534
* #531
* #532

### CODE REVIEW

**Comment by saqadri** [comment]
Is this diff ready for review?

File: extensions/Gemini/python/src/aiconfig_extension_gemini/Gemini.py
Comment (rossdanlm) [inline extensions/Gemini/python/src/aiconfig_extension_gemini/Gemini.py:134]: Nit: add doc of API: https://ai.google.dev/tutorials/python_quickstart#multi-turn_conversations

**Review by rossdanlm — APPROVED** [review]
nice, pls don't land until #3 is also approved!

**Pull request merged**
